packages/deployer/src/cli/commands.ts

```diff
@@ -40,6 +40,14 @@ export async function deploy(
     throw new Error('Invalid mnemonic');
   }
 
+  let apiKey: string | undefined = undefined;
+  if (config.nodeSettings.enableTestingGateway) {
+    apiKey = secrets.ENDPOINT_TESTING_API_KEY;
+    if (!apiKey) {
+      throw new Error('Unable to deploy testing gateway as the API key is missing');
```

Comment:
Maybe put the key in the message just as an additional tip to the user?
```suggestion
      throw new Error('Unable to deploy testing gateway as the ENDPOINT_TESTING_API_KEY secret is missing');
```